File: docs/breez-sdk/snippets/go/messages.go
package example

import (
	"errors"
	"log"

	"github.com/breez/breez-sdk-spark-go/breez_sdk_spark"
)

func SignMessage(sdk *breez_sdk_spark.BreezSdk) (*breez_sdk_spark.SignMessageResponse, error) {
	// ANCHOR: sign-message
	message := "<message to sign>"
	// Set to true to get a compact signature rather than a DER
	compact := true

	signMessageRequest := breez_sdk_spark.SignMessageRequest{
		Message: message,
		Compact: compact,
	}
	signMessageResponse, err := sdk.SignMessage(signMessageRequest)
	if err != nil {
		var sdkErr *breez_sdk_spark.SdkError
		if errors.As(err, &sdkErr) {
			// Handle SdkError - can inspect specific variants if needed
			// e.g., switch on sdkErr variant for InsufficientFunds, NetworkError, etc.
		}
		return nil, err
	}

	signature := signMessageResponse.Signature
	pubkey := signMessageResponse.Pubkey

	log.Printf("Pubkey: %v", pubkey)
	log.Printf("Signature: %v", signature)
	// ANCHOR_END: sign-message
	return &signMessageResponse, nil
}

func CheckMessage(sdk *breez_sdk_spark.BreezSdk) (*breez_sdk_spark.CheckMessageResponse, error) {
	// ANCHOR: check-message
	message := "<message>"
	pubkey := "<pubkey of signer>"
	signature := "<message signature>"

	checkMessageRequest := breez_sdk_spark.CheckMessageRequest{
		Message:   message,
		Pubkey:    pubkey,
		Signature: signature,
	}
	checkMessageResponse, err := sdk.CheckMessage(checkMessageRequest)
	if err != nil {
		var sdkErr *breez_sdk_spark.SdkError
		if errors.As(err, &sdkErr) {
			// Handle SdkError - can inspect specific variants if needed
			// e.g., switch on sdkErr variant for InsufficientFunds, NetworkError, etc.
		}
		return nil, err
	}

	isValid := checkMessageResponse.IsValid

	log.Printf("Signature valid: %v", isValid)
	// ANCHOR_END: check-message
	return &checkMessageResponse, nil
}

File: docs/breez-sdk/snippets/rust/src/messages.rs
use anyhow::Result;
use breez_sdk_spark::*;
use tracing::info;

#[allow(dead_code)]
async fn sign_message(sdk: &BreezSdk) -> Result<()> {
    // ANCHOR: sign-message
    let message = "<message to sign>".to_string();
    // Set to true to get a compact signature rather than a DER
    let compact = true;

    let sign_message_request = SignMessageRequest { message, compact };
    let sign_message_response = sdk.sign_message(sign_message_request).await?;

    let signature = sign_message_response.signature;
    let pubkey = sign_message_response.pubkey;

    info!("Pubkey: {}", pubkey);
    info!("Signature: {}", signature);
    // ANCHOR_END: sign-message
    Ok(())
}

#[allow(dead_code)]
async fn check_message(sdk: &BreezSdk) -> Result<()> {
    // ANCHOR: check-message
    let check_message_request = CheckMessageRequest {
        message: "<message>".to_string(),
        pubkey: "<pubkey of signer>".to_string(),
        signature: "<message signature>".to_string(),
    };
    let check_message_response = sdk.check_message(check_message_request).await?;

    let is_valid = check_message_response.is_valid;

    info!("Signature valid: {}", is_valid);
    // ANCHOR_END: check-message
    Ok(())
}
